src/Conversion/ONNXToTOSA/DialectBuilder.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//====------ DialectBuilder.hpp - TOSA dialect builder --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file contains the dialect build for the TOSA dialect. Uses the same
+// implementation as ONNXToMhlo with minor differences.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+
+#include "src/Conversion/ONNXToTOSA/DialectBuilder.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+// =============================================================================
+// IndexExpr Builder for Lowering using Shape/TOSA Dialect.
```

Comment:
Great to see that you are trying to use the new approach.

Note that if you find it hard to use the TOSA dialect to analyze the code, you always have the option (during the ONNX to TOSA) to look at the old (onnx, before conversion) instruction stream. 

To get the old, get the shape helper with the operands that are attached directly to the op that is being analyzed. This is done by constructing `ONNXxxxOpShapeHelper shapeHelper(op, {});`, namely empty operands. To get the new, build the shape helper with the operand ranged passed to the conversion function.

I did not have experience with TOSA, but when I built something alike for MHLO, there were indeed a large stream of different operations, so I sometimes used the old/before conversion ONNX operation stream.

Reply:
I will keep this in mind, thank you!